include/sparrow/layout/variable_size_binary_array.hpp

```diff
@@ -570,4 +662,211 @@ namespace sparrow
+        const auto pos_to_insert = sparrow::next(data_buffer.cbegin(), offset_begin);
+        data_buffer.insert(pos_to_insert, joined_repeated_value_range.begin(), joined_repeated_value_range.end());
+        insert_offset(offsets_cbegin() + idx + 1, static_cast<offset_type>(value.size()), count);
+        return sparrow::next(value_begin(), idx);
+    }
+
+    template <std::ranges::sized_range T, class CR, layout_offset OT>
+    auto variable_size_binary_array<T, CR, OT>::insert_offset(
+        const_offset_iterator pos,
+        offset_type value_size,
+        size_type count
+    ) -> offset_iterator
+    {
+        auto& offset_buffer = get_arrow_proxy().get_array_private_data()->buffers()[OFFSET_BUFFER_INDEX];
+        const auto idx = static_cast<size_t>(std::distance(offsets_cbegin(), pos));
+        auto offset_buffer_adaptor = make_buffer_adaptor<OT>(offset_buffer);
+        const offset_type cumulative_size = value_size * static_cast<offset_type>(count);
+        // Adjust offsets for subsequent elements
+        std::for_each(
+            sparrow::next(offset_buffer_adaptor.begin(), idx + 1),
+            offset_buffer_adaptor.end(),
+            [cumulative_size](auto& offset)
+            {
+                offset += cumulative_size;
+            }
+        );
+        offset_buffer_adaptor.insert(sparrow::next(offset_buffer_adaptor.cbegin(), idx + 1), count, 0);
+        // Put the right values in the new offsets
+        for (size_t i = idx + 1; i < idx + 1 + count; ++i)
+        {
+            offset_buffer_adaptor[i] = offset_buffer_adaptor[i - 1] + value_size;
+        }
+        return offsets_begin() + idx;
+    }
+
+    template <std::ranges::sized_range T, class CR, layout_offset OT>
+    template <mpl::iterator_of_type<T> InputIt>
+    auto
+    variable_size_binary_array<T, CR, OT>::insert_values(const_value_iterator pos, InputIt first, InputIt last)
+        -> value_iterator
+    {
+        auto& data_buffer = get_arrow_proxy().get_array_private_data()->buffers()[DATA_BUFFER_INDEX];
+        const auto idx = static_cast<size_t>(std::distance(value_cbegin(), pos));
+        const OT offset_begin = *offset(idx);
+        auto values = std::ranges::subrange(first, last);
+        auto joined_range = values | std::ranges::views::join;
+        auto casted_joined_range = joined_range
+                                   | std::views::transform(
+                                       [](char c) -> uint8_t
+                                       {
+                                           return static_cast<uint8_t>(c);
+                                       }
+                                   );
+        const auto insert_pos = sparrow::next(data_buffer.cbegin(), offset_begin);
+        std::vector<uint8_t> casted_values;
+        std::ranges::copy(casted_joined_range, std::back_inserter(casted_values));
+        data_buffer.insert(insert_pos, casted_values.begin(), casted_values.end());
```

Comment:
I'm not satisfied of this part of the code. I have a temporary vector to be able to insert the data in the buffer.
I failed to avoid that, I need help on that part

Reply:
Fixed by just iterating and copying the values